### PR TITLE
Clean up and document Gradle Managed Devices

### PR DESCRIPTION
**Pull request by timothyfroehlich**
See the additions to the readme, but the short version is to run `./gradlew gmdTestQuick` to run all instrumented tests quickly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -142,6 +142,24 @@ To update the Design Switcher, temporarily set the `DISABLE_LIVE_MODE` flag in `
...
 
+## Running Instrumented Tests
+
+All instrumented tests can be run on a running emulator by running "./gradlew connectedCheck". Additionally they can be run on [Gradle Managed Devices](https://developer.android.com/studio/test/gradle-managed-devices). These devices are the standard test targets and all tests must pass on them prior to a release. They are collected into three gradle tasks:
+
+- `gmdTestsQuick` run the tests on an [ATD image](https://developer.android.com/studio/test/gradle-managed-devices#gmd-atd), which is optimized for instrumented tests.
+- `gmdTestsStandard` runs the tests on both the ATD and the most current Android image
+- `gmdTestsAll` runs the tests on all configured Gradle Managed devices,including the above and any additional APIs that have been chosen to be tested against.
```
Comment (rylin8) [inline README.md:151]: These should be `gmdTestQuick`, etc with a singular `Test` in the name.

**Comment by rylin8** [comment]
First time running, ran into this:
```
Execution failed for task ':designcompose:gmdApi33Setup'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.ManagedDeviceInstrumentationTestSetupTask$ManagedDeviceSetupRunnable
   > java.lang.IllegalStateException: Snapshot setup ran successfully, but the snapshot failed to be created. This is
     likely to a lack of disk space for the snapshot. Try the cleanManagedDevices
     task with the --unused-only flag to remove any unused devices for this project.
 ```

**Comment by timothyfroehlich** [comment]
> First time running, ran into this:
> 
> ```
> Execution failed for task ':designcompose:gmdApi33Setup'.
> > A failure occurred while executing com.android.build.gradle.internal.tasks.ManagedDeviceInstrumentationTestSetupTask$ManagedDeviceSetupRunnable
>    > java.lang.IllegalStateException: Snapshot setup ran successfully, but the snapshot failed to be created. This is
>      likely to a lack of disk space for the snapshot. Try the cleanManagedDevices
>      task with the --unused-only flag to remove any unused devices for this project.
> ```

Did you try what the error suggested?

**Comment by rylin8** [comment]
I cleaned then ran the quick test, and got:
```
> Task :tutorial-app:mergeDebugAndroidTestTestResultProtos
Test execution completed. See the report at: file:///data/src/automotive-design-compose/reference-apps/tutorial/app/build/reports/androidTests/managedDevice/debug/allDevices/index.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tutorial-app:gmdAtdApi30DebugAndroidTest'.
> Computer Architecture Environment not initialized properly.
```

**Comment by timothyfroehlich** [comment]
I suspect there's something wrong with your system, hopefully it's an easy fix. Let's jump on a chat to figure it out when you have time.

**Comment by rylin8** [comment]
@iamralpht could you try running `./gradlew gmdTestQuick` when you get a chance? I spent some time with Tim and we were unable to get this to work on my machine.

**Review by rylin8 — APPROVED** [review]

**Pull request merged**